app/utils/testing.py

```diff
@@ -0,0 +1,55 @@
+from config.api import handle_404_not_found
+from config.api import handle_django_validation_error
+from config.api import handle_exception
+from config.api import handle_http_error
+from config.api import handle_ninja_validation_error
+from config.api import handle_unauthorized
+from ninja import NinjaAPI
+from ninja import Router
+from ninja.errors import AuthenticationError
+from ninja.errors import HttpError
+from ninja.errors import ValidationError as NinjaValidationError
+from ninja.testing import TestClient as BaseTestClient
+
+from django.core.exceptions import ValidationError
+from django.http import Http404
+
+
+class TestClient(BaseTestClient):
+    """ A patched version of the ninja test client which re-attaches the custom exception handlers.
+    This allows using the custom exception handling in tests.
+
+    """
+
+    def __init__(
+        self,
+        router_or_app: NinjaAPI | Router,
+        headers: dict[str, str] | None = None,
+    ) -> None:
+        super().__init__(router_or_app=router_or_app, headers=headers)
+
+        # In order to be able to test exception handling, we need to attach the
```

Comment:
I'm wondering if it would've been easier to stop using the NinjaAPI test client and instead use the more general django one. Then you have the full call stack and test the request as it really would be coming from external.

Reply:
Maybe add this to the backlog of improvements as it is currently working like this?

Reply:
As discussed in person: Add this to the stack of technical dept.